### PR TITLE
UI-432 Updates to Input to match coding standards

### PR DESCRIPTION
**Pull request by joshsylvester**
* added new InputField component internally
* added simple test files, passing.
* broke out code to match current coding standards.
* ensured existing demos in storybook still worked as expected.

NOTE:
The "Input w/ icon to the left & right" does not render properly in storybook because the Input.scss file is not being included properly. [UI-605](https://servicemax.atlassian.net/browse/UI-605) has been opened to address the issue.

### CODE REVIEW

File: lib/components/Input/Input.js [hunk 1]
```diff
@@ -1,172 +1,110 @@
...
+  ]),
+  inputRef: PropTypes.func,
+  onChange: PropTypes.func,
+  onKeyDown: PropTypes.func,
+  placeholder: PropTypes.string,
+  readOnly: PropTypes.bool,
+  symbolPattern: PropTypes.string,
+  type: PropTypes.string,
+  value: PropTypes.string,
+};
```
[the remark this answers is not in the document]
Comment (joshsylvester) [reply lib/components/Input/Input.js]: good call - i wasn't building with the forward thinking toward docgen. i'll add those today.

File: lib/components/Input/Input.js [hunk 2]
```diff
@@ -1,172 +1,110 @@
...
-          { addonRight ? this.renderAddon(addonRight) : undefined }
-        </div>
-      );
+    } else {
+      inputClassName = classnames('Input');
     }
-    return this.renderInput(inputProps);
+    return (
+      <div className={inputClassName}>
+        { addonLeft ? this.renderAddon(addonLeft) : undefined }
```
[the remark this answers is not in the document]
Comment (joshsylvester) [reply lib/components/Input/Input.js]: great call-out. i'll update them to use the better syntax.

**Comment by joshsylvester** [comment]
I made a few big updates to the input component, @mallik-svmx @sheela-svmx please review.
* updated the internal `InputField` to be a controlled component
* added onBlur and onFocus function pass throughs
* cleaned up the manual adding of props, allowed the `{..inputProps}` to pass them
* removed the defaultValue as it wasn't used anymore
* added propType comments for storybook/docgen
* updated stories to match changes

File: lib/components/Input/Input.js [hunk 3]
```diff
@@ -1,172 +1,164 @@
...
+      inputClassName = classnames('Input');
     }
-    return this.renderInput(inputProps);
+    return (
+      <div className={inputClassName}>
+        {addonLeft && this.renderAddon(addonLeft)}
+        {iconLeft && this.renderIcon(iconLeft, 'left')}
+        <InputField {...inputProps} />
+        {iconRight && this.renderIcon(iconRight, 'right')}
+        {addonRight && this.renderAddon(addonRight)}
```
Comment (sheela-svmx) [inline lib/components/Input/Input.js]: Should these icons inside input field be looked like Buttons where the <Icon> is passed into the Input component if they have 

```
<Input>
  <Icon align='right' />
</Input>
```

Segregating Icon related things away from the <Input> component.

Comment (joshsylvester) [reply lib/components/Input/Input.js]: in this case we're leaving it separate for a few reasons:
1. the icons are intended to render dynamically
2. we're dealing with more than one, 
3. and the Icon is not "inside" the `Input`, but rendered alongside it, somewhat as a helper.. 
4. an `Input` field doesn't really have children - it has a value and that value *MUST* be a string.

with `Button`, the `Icon` is actually rendered inside the other component as one of it's children, so it makes sense to manually render it as you expect - especially in the case where i want to render a button with an icon and text, and want control over how they render together... it's easier to handle that case (with `Button`) just using children, without building all the logic on how to handle a button with text, icon, icon with text, icon on the right with text, text icon text... etc... it'll just get confusing.

In the case where we'd be rendering the icon / text / whatnot outside the button as another component (ie: InputGroup) - we may take all this out and end up having something like this:

```
<InputGroup>
  <Text>addon</Text>
  <Icon align="left" name="arrow-left" />
  <Input value="123456789" name="input-field" />
  <Icon align="right" name="arrow-right" />
</InputGroup>
```

which is a direction that we may consider down the line - the path currently used was the least "disruptive" to the current flow. Ideally i'd like them both _predix_ and _lightning_to use the `InputGroup` paradigm and remove this extra icon / addon features and leave the component "pure".

Comment (sheela-svmx) [reply lib/components/Input/Input.js]: Makes Sense cool

**Pull request merged**